assets/src/edit-story/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,171 @@
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@web-stories-wp/i18n';
+import { Input } from '@web-stories-wp/design-system';
+import { useState, useCallback } from 'react';
+import styled from 'styled-components';
+import {
+  createResource,
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+  getImageDimensions,
+  getVideoDimensions,
+} from '@web-stories-wp/media';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../../app';
+import Dialog from '../../../../../dialog';
+import useLibrary from '../../../../useLibrary';
+import {
+  useUploadVideoFrame,
+  getPosterName,
+} from '../../../../../../app/media/utils';
+import { withProtocol } from '../../../../../../utils/url';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const { allowedFileTypes } = useConfig();
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+
+  const [errorMsg, setErrorMsg] = useState(false);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
+    description
+  );
+  const [link, setLink] = useState('');
+
+  const getFileInfo = useCallback(
+    (value = link) => {
+      const ext = value.split(/[#?]/)[0].split('.').pop().trim();
+      if (!allowedFileTypes.includes(ext)) {
+        setErrorMsg(error);
+      } else {
+        setErrorMsg(null);
+      }
+      return {
+        ext,
+        type: ['m4v', 'mp4'].includes(ext) ? 'video' : 'image',
+      };
+    },
+    [link, allowedFileTypes, error]
+  );
+
+  const onInsert = useCallback(async () => {
+    if (errorMsg?.length) {
+      return;
+    }
+    try {
+      const { type } = getFileInfo();
+      const isVideo = type === 'video';
+      const getMediaDimensions = isVideo
+        ? getVideoDimensions
+        : getImageDimensions;
+      const { width, height } = await getMediaDimensions(link);
+
+      // Create a poster file in case of videos.
+      let posterData;
+      if (isVideo) {
+        const originalFileName = getFileNameFromUrl(link);
+        const fileName = getPosterName(originalFileName);
+        const posterFile = await getFirstFrameOfVideo(link);
+        posterData = await uploadVideoPoster(0, fileName, posterFile);
```

Comment:
What happens here if user can not upload files?

Reply:
Hm, good question. Should we ask for a link for the poster image for a video then as well in this case, and not allow adding it otherwise? Or not allow inserting videos from external link at all for these users?

Reply:
IIRC we don't want/need links for poster image for now, so we should simply not try to upload the video poster if the user has no permission.

We can still allow contributors to hotlink though. It's just that the poster would need to be added by someone else who has permission.

Details can be discussed in a new ticket